### PR TITLE
core/service: store BUSERROR= & VARLINKERROR= received and show them through systemctl status

### PR DESCRIPTION
**Pull request by YHNdnzj**
Fixes: #6073

### CODE REVIEW

File: src/libsystemd/sd-bus/bus-error.c [hunk 1]
```diff
@@ -78,8 +76,10 @@ static int bus_error_name_to_errno(const char *name) {
                 return r;
         }
 
+        const sd_bus_error_map *m;
```
Comment (poettering) [inline src/libsystemd/sd-bus/bus-error.c]: i'd probably nove this into the for() loop below where it is used for the first time, and then declare it a 2nd type for the ALIGN_PTR() assignment further down.

**Comment by poettering** [comment]
love it. thanks!

**Review by yuwata — APPROVED** [review]
LGTM. Is it possible to add simple test cases?

**Comment by poettering** [comment]
for writing a test case, consider just cherry-picking #33434 into this PR (and closing that PR then), and then fire off a `systemd-run -p Notify=main busctl call …` command line that calls something that fails (could be as easy as a service or method that definitely doesn exist), and see if systemctl status later reports it.

File: man/org.freedesktop.systemd1.xml [hunk 2]
```diff
@@ -2745,6 +2745,8 @@ node /org/freedesktop/systemd1/unit/avahi_2ddaemon_2eservice {
       readonly s FileDescriptorStorePreserve = '...';
```
Comment (poettering) [inline man/org.freedesktop.systemd1.xml:2745]: btw, downstreams asked us to always make changes to TODO in separate commits because they often backport patches and the TODO stuff is a major source of conflicts because it changes so often, but not what they care about for backports.

**Comment by poettering** [comment]
lgtm.

File: man/org.freedesktop.systemd1.xml (hunk 2 above)
Comment (YHNdnzj) [reply man/org.freedesktop.systemd1.xml:2745]: Hmm, TODO is mostly about new features, so I don't quite understand why they would backport those commits in the first place... I prefer not to split such changes out if this is only about backporting.

Comment (poettering) [reply man/org.freedesktop.systemd1.xml:2745]: well, people backport stuff for many reasons. i personally would be a lot more conservative, but then again i am not a package maintainer.
